workspaces/adventure-pack/goodies/typescript/Iterator.prototype.toSet/index.test.ts

```diff
@@ -0,0 +1,113 @@
+import { describe, expect, it } from "@jest/globals";
+
+import "../Iterator.prototype.filter";
+import "../Iterator.prototype.map";
+
+import { iteratorPrototype } from "../Iterator.prototype";
+delete (iteratorPrototype as unknown as Record<string, unknown>).toSet;
+// eslint-disable-next-line import-x/first -- This has to happen after we delete the built-in implementation.
+import "./index";
+
+describe("Iterator.prototype.toSet", () => {
+  it("adds primitives to a Set", () => {
+    const arr = [1, 2, 2, 3, 1, 2, 3, 3, 100, false, 100, false, "hello"];
+    const set = arr.values().toSet();
+
+    expect(set).toStrictEqual(new Set([1, 2, 3, 100, false, "hello"]));
+  });
+
+  it("adds objects to a Set", () => {
```

Comment:
I think a key part of this test as well as the one below is that we respect referential equality of objects (and arrays). I'd express that in the title of the test, so that it's clear that this was part of the intent.

So for example:

```ts
it("respects referential equality of objects", () => {
```

Reply:
that sounds good, I like it! thank you